mypy/sametypes.py

```diff
@@ -25,7 +25,6 @@ def is_same_type(left: Type, right: Type) -> bool:
         # types can be simplified to non-union types such as Union[int, bool]
         # -> int. It would be nice if we always had simplified union types but
         # this is currently not the case, though it often is.
-        left = simplify_union(left)
```

Comment:
Unfortunately this doesn't work for cases like checking whether `Union[A, B]` is the same type as `Union[B, A]`, where `B` is a subclass of `A`.

One possible way to fix this would be to use a simpler replacement for `is_same_type` in `is_protocol_implementation`. Maybe we can just use type equality (==)?

Reply:
Thank you for the feedback!

Ah, my assumption is indeed wrong. Unfortunately I don't know how I could integrate that as a test for mypy. Could you help me get one, so I can be sure any other changes I make don't regress this?

Oh, I previously tried something similar - copying `is_same_type` to `subtypes` with my change but that didn't work. However just using `==` seems to work fine. I'll commit this soon.

Reply:
You can write unit tests for `is_same_type` in `mypy/test/testtypes.py`. `TypeFixture` contains types that you can use as union type items.

Reply:
Thank you. I should be able to get a new unit test committed tomorrow.